### ISSUE TITLE
no_space_in_method_call rule fails with trailing closures in Swift 5.6 (Xcode 13.3)

### ISSUE THREAD

**Issue by eneko** [realm/SwiftLint#3836]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

Note: I haven't tested this issue on the latest version of SwiftLint. However, looking at the release log, I couldn't find any mentions of this issue being fixed.

### Describe the bug

When running SwiftLint on a code base where trailing closures are used, and the `no_space_in_method_call` rule is enabled, we get false-positives:

```swift
let foo = bar.flatMap { $0 }  // <-- this will trigger the rule in Swift 5.6
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.39.2
* Which Xcode version are you using (check `xcodebuild -version`)? 13.3 (beta) 
* Do you have a sample that shows the issue? Run `echo "let foo = bar.flatMap { baz in baz }" | swiftlint lint --no-cache --use-stdin --enable-all-rules` to quickly test if your example is really demonstrating the issue. 
```swift
// This triggers a violation in Swift 5.6:
let foo = bar.flatMap { $0 }
```

**Comment by marcelofabri** [comment]
This doesn't happen on the latest version of SwiftLint due to https://github.com/realm/SwiftLint/pull/3280, which was released in [0.40.0](https://github.com/realm/SwiftLint/releases/tag/0.40.0)
